### DISCUSSION TITLE
Scaling and units on vector output (SVG/PDF)

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#226] [Q&A] [answered]
I have been using "magic" hard-coded "scale factors" when outputting SVG or PDF files.

Is there a better way of getting consistent  "mm" or "inches" sizes for a sketch output?

**Comment by hx2A** [comment, marked as answer]
I don't know about PDF files but I know for SVG you can do something like this

```xml
<svg width="130mm" height="178mm" viewBox="0 0 130 178"
xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
xmlns="http://www.w3.org/2000/svg"
xmlns:svg="http://www.w3.org/2000/svg">
```

for a Sketch with size width of 130 and height of 178. Inserting that other stuff after the`svg` tag will make it so that if you open it with Inkscape or plot it with a plotter, the dimensions of the shapes drawn will be the same as designed in the code.
